snakemake/hla.rules

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2018. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+HLA typing and gene quantification.
+
+TODO: replace this with allele-specific quantification that works
+across multiple species.
+"""
+
+rule seq2hla_paired_end:
+  input:
+    r1 = "{prefix}_R1.fastq.gz",
+    r2 = "{prefix}_R2.fastq.gz",
+  output:
+    "{prefix}-ClassI.expression",
+    "{prefix}-ClassII.expression",
+    "{prefix}-ClassI.HLAgenotype4digits",
+    "{prefix}-ClassII.HLAgenotype4digits",
+  resources:
+    mem_mb = _mem_gb_for_alignment() * 1024
+  benchmark:
+    "{prefix}_seq2hla_benchmark.txt"
+  log:
+    "{prefix}_seq2hla.log"
+  threads: _get_threads_for_alignment
```

Comment:
ditto here re: using alignment-related resources. i default to half-cores or all-cores for expensive parallelizable tasks. e.g. check out the `dna_indel_realigner_target_creator_per_chr` rule in `gatk.rules` to see how memory/thread resources work there